prow/plugins/lgtm/lgtm.go

```diff
@@ -314,12 +335,53 @@ func handlePullRequest(gc ghLabelClient, pe github.PullRequestEvent, log *logrus
 		return nil
 	}
 
-	// Don't bother checking if it has the label... it's a race, and we'll have
-	// to handle failure due to not being labeled anyway.
 	org := pe.PullRequest.Base.Repo.Owner.Login
 	repo := pe.PullRequest.Base.Repo.Name
 	number := pe.PullRequest.Number
 
+	// rc.repo.Owner.Login, rc.repo.Name)
+	opts := optionsForRepo(config, org, repo)
+	if opts.StoreTreeHash {
+		// Check if we have LGTM label
+		labels, err := gc.GetIssueLabels(org, repo, number)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to get labels.")
+		}
+
+		if github.HasLabel(LGTMLabel, labels) {
+			// Check if we have a tree-hash comment
+			var lastLgtmTreeHash string
+			botname, err := gc.BotName()
+			if err != nil {
+				return err
+			}
+			comments, err := gc.ListIssueComments(org, repo, number)
+			if err != nil {
+				log.WithError(err).Errorf("Failed to get issue comments.")
+			}
+			for _, comment := range comments {
+				m := addLGTMLabelNotificationRe.FindStringSubmatch(comment.Body)
+				if comment.User.Login == botname && m != nil && comment.UpdatedAt.Equal(comment.CreatedAt) {
+					lastLgtmTreeHash = m[1]
+					break
+				}
+			}
+			// Get the current tree-hash
+			if pe.PullRequest.MergeSHA != nil {
```

Comment:
This check could go before `gc.BotName` and `gc.ListIssueComments` to save API tokens.